skbio/sequence/_substitution.py

```diff
@@ -0,0 +1,518 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy as np
+
+from skbio.util._decorator import experimental, classonlymethod
+from skbio.stats.distance import DissimilarityMatrix
+
+
+@experimental(as_of='0.5.10')
+class SubstitutionMatrix(DissimilarityMatrix):
```

Comment:
Are there examples of substitution matrices which are not symmetric? If symmetry is expected, then I believe this should use `DistanceMatrix`

Reply:
The common matrices provided by NCBI are all symmetric (I checked). However, asymmetric substitution matrices do exist, although rare (see [here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7586916/) for examples). There could be a reason why NCBI matrices document both upper and lower triangles, and why other packages like biotite support asymmetric matrices (see [here](https://www.biotite-python.org/apidoc/biotite.sequence.align.SubstitutionMatrix.html?highlight=substitutionmatrix#biotite.sequence.align.SubstitutionMatrix.is_symmetric)). I think that one advantage of allowing asymmetric matrices is that the researchers can test specific hypotheses of evolution and mutation in a highly controlled setting (e.g., positively charged amino acids have a selective advantage at a catalytic site, therefore one a site become positively charged, it won't go back).

 Some other packages